<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<HTML>
 <HEAD>
   <TITLE> [Twisted-Python] If the errbacks of a canceled Deferred are called with error other than CancelledError, is this acceptable?
   </TITLE>
   <LINK REL="Index" HREF="index.html" >
   <LINK REL="made" HREF="mailto:twisted-python%40twistedmatrix.com?Subject=Re%3A%20%5BTwisted-Python%5D%20If%20the%20errbacks%20of%20a%20canceled%20Deferred%20are%0A%20called%20with%20error%20other%20than%20CancelledError%2C%20is%20this%20acceptable%3F&In-Reply-To=%3C20130620193435.10961.1679313466.divmod.xquotient.30%40top%3E">
   <META NAME="robots" CONTENT="index,nofollow">
   <style type="text/css">
       pre {
           white-space: pre-wrap;       /* css-2.1, curent FF, Opera, Safari */
           }
   </style>
   <META http-equiv="Content-Type" content="text/html; charset=utf-8">
   <LINK REL="Previous"  HREF="059576.html">
   <LINK REL="Next"  HREF="059582.html">
 </HEAD>
 <BODY BGCOLOR="#ffffff">
   <H1>[Twisted-Python] If the errbacks of a canceled Deferred are called with error other than CancelledError, is this acceptable?</H1>
    <B>exarkun at twistedmatrix.com</B> 
    <A HREF="mailto:twisted-python%40twistedmatrix.com?Subject=Re%3A%20%5BTwisted-Python%5D%20If%20the%20errbacks%20of%20a%20canceled%20Deferred%20are%0A%20called%20with%20error%20other%20than%20CancelledError%2C%20is%20this%20acceptable%3F&In-Reply-To=%3C20130620193435.10961.1679313466.divmod.xquotient.30%40top%3E"
       TITLE="[Twisted-Python] If the errbacks of a canceled Deferred are called with error other than CancelledError, is this acceptable?">exarkun at twistedmatrix.com
       </A><BR>
    <I>Thu Jun 20 13:34:35 MDT 2013</I>
    <P><UL>
        <LI>Previous message (by thread): <A HREF="059576.html">[Twisted-Python] If the errbacks of a canceled Deferred are	called with error other than CancelledError, is this acceptable?
</A></li>
        <LI>Next message (by thread): <A HREF="059582.html">[Twisted-Python] If the errbacks of a canceled Deferred are called with error other than CancelledError, is this acceptable?
</A></li>
         <LI> <B>Messages sorted by:</B> 
              <a href="date.html#59581">[ date ]</a>
              <a href="thread.html#59581">[ thread ]</a>
              <a href="subject.html#59581">[ subject ]</a>
              <a href="author.html#59581">[ author ]</a>
         </LI>
       </UL>
    <HR>  
<!--beginarticle-->
<PRE>On 19 Jun, 07:49 pm, <A HREF="https://twistedmatrix.com/cgi-bin/mailman/listinfo/twisted-python">tom.prince at ualberta.net</A> wrote:
&gt;<i>
</I>&gt;<i>Well, it already raises `CancelledError`. Every deferred that doesn't
</I>&gt;<i>have explicitly handle cancelling already does:
</I>&gt;<i>
</I>&gt;<i>from twisted.mail import smtp
</I>&gt;<i>from twisted.python import log
</I>&gt;<i>
</I>&gt;<i>d = smtp.sendmail(&quot;host&quot;, &quot;options.sender&quot;, [], &quot;&quot;)
</I>&gt;<i>d.cancel()
</I>&gt;<i>d.addErrback(log.err, &quot;Here be CancelledError&quot;)
</I>
Thank you for pointing this out.  It seems like an important fact that 
makes the rest of the discussion moot.

By making `Deferred.cancel` work on any Deferred by triggering a 
`CancelledError`, we have already decided on the failure behavior for 
all existing Deferred-returning APIs.  Changing that at this point 
doesn't seem like a very good idea.

I think I'd also like to challenge the idea that Glyph put forwards 
earlier in the thread that this extra information is *necessarily* 
important to the application.

So far, I haven't written any applications that care about the exact 
stage at which the operation is cancelled.  All they care about is that 
the operation *is* cancelled (ie, resources are cleaned up) and that the 
Deferred fires soon.  There's lots of utility in just this level of 
functionality which requires no extra information about the internal 
progress of the operation.

This is not to say that I believe there is no application that might 
want this information, but maybe someone can propose some concrete use 
cases for this information and design can follow from that.  So far I 
don't think any practical justification to do anything other than 
`CancelledError` has been presented.

Jean-Paul


</PRE>

<!--endarticle-->
    <HR>
    <P><UL>
        <!--threads-->
	<LI>Previous message (by thread): <A HREF="059576.html">[Twisted-Python] If the errbacks of a canceled Deferred are	called with error other than CancelledError, is this acceptable?
</A></li>
	<LI>Next message (by thread): <A HREF="059582.html">[Twisted-Python] If the errbacks of a canceled Deferred are called with error other than CancelledError, is this acceptable?
</A></li>
         <LI> <B>Messages sorted by:</B> 
              <a href="date.html#59581">[ date ]</a>
              <a href="thread.html#59581">[ thread ]</a>
              <a href="subject.html#59581">[ subject ]</a>
              <a href="author.html#59581">[ author ]</a>
         </LI>
       </UL>

<hr>
<a href="https://twistedmatrix.com/cgi-bin/mailman/listinfo/twisted-python">More information about the Twisted-Python
mailing list</a><br>
</body></html>
